### PR TITLE
fix the bugs in interrupt handling and mpu buffer validate

### PR DESCRIPTION
**Pull request by vonhust**
This PR fixes #22288

It includes two parts
    * the bug in interrupt handling when interrupt comes out in user mode when USER_SPACE is configured. The root cause it that the switch between aux_user_sp/aux_sec_k_sp and sp relies on 
the correct vaule of aux_irq_act' U bit.
    * the bug in mpu v3' driver, the access to mpu regs should be atomic, mpu buffer validate is not procted by a lock, other mpu apis are protected by high level zephyr API, e.g. mem domain.

With this PR, the test in #22288 is passed.

### CODE REVIEW

File: arch/arc/core/mpu/arc_mpu_v3_internal.h [hunk 1]
```diff
@@ -656,7 +656,7 @@ int arc_core_mpu_get_max_domain_partition_regions(void)
 int arc_core_mpu_buffer_validate(void *addr, size_t size, int write)
 {
 	int r_index;
-
+	int key = irq_lock();
```
Comment (andyross) [inline arch/arc/core/mpu/arc_mpu_v3_internal.h]: You probably want arch_irq_lock() here if you're just checking local CPU state, or a spinlock if the state is visible from other CPUs.  Remember that in SMP, irq_lock() is a somewhat heavyweight **global** lock that will contend with any irq_lock() user anywhere in the system.

Comment (vonhust) [reply arch/arc/core/mpu/arc_mpu_v3_internal.h]: thanks your comment, it make sense, i will modify it

**Review by abrodkin — APPROVED** [review]
Looks good to me.

**Comment by ruuddw** [comment]
Looks ok to me, just wondering if the v2 MPU buffer validate requires a some locking as well?

**Comment by andrewboie** [comment]
This PR has my patch https://github.com/zephyrproject-rtos/zephyr/pull/22393/commits/c507478007eb1d2666a5c64ee2bff5a6b3396418 included, this is in another PR and can't be merged until ARM is fixed, please remove it, it's probably why this is failing CI

**Comment by vonhust** [comment]
> Looks ok to me, just wondering if the v2 MPU buffer validate requires a some locking as well?

I review the MPUv2's code and run the test,  it does not need a lock as it has no index reg and the access to reg is atomic.

**Comment by ruuddw** [comment]
> > Looks ok to me, just wondering if the v2 MPU buffer validate requires a some locking as well?
> 
> I review the MPUv2's code and run the test, it does not need a lock as it has no index reg and the access to reg is atomic.

Thanks for checking, so the atomicity issue was only for 'read-only' operations (with a conflict on the index register based MPU v2 programming), and not for operations changing the MPU regions themselves during the MPU buffer validate? I assume that is somehow protected on a higher level (or, we don't have a test-case showing the race condition...).

**Review by ruuddw — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request merged**